### PR TITLE
SAA-221 Create Activities Management API location endpoints

### PR DESCRIPTION
**Pull request by markford-moj**
New activities management api : Get location prefix by group (migrated from whereabouts API)

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/LocationPrefixIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      .accept(MediaType.APPLICATION_JSON)
+      .headers(setAuthorisation(roles = listOf()))
+      .exchange()
+      .expectStatus().isNotFound
+      .expectHeader().contentType(MediaType.APPLICATION_JSON)
+      .expectBody(ErrorResponse::class.java)
+      .returnResult().responseBody
+
+    with(errorResponse!!) {
+      Assertions.assertThat(errorCode).isNull()
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/LocationPrefixIntegrationTest.kt:52]: Tidy up Assertions import, meh.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/LocationPrefixIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    val groupName = "Houseblock 1"
+
+    webTestClient.get()
+      .uri { uriBuilder: UriBuilder ->
+        uriBuilder
+          .path("/prisons/{prisonCode}/location-prefix")
+          .queryParam("groupName", groupName)
+          .build(prisonCode)
+      }
+      .headers(setAuthorisation(roles = listOf("ROLE_LICENCE_CA", "ROLE_KW_ADMIN")))
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/LocationPrefixIntegrationTest.kt:24]: Should these roles be here?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationPrefixControllerTest.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    whenever(
+      locationService.getLocationPrefixFromGroup(
+        "MDI", "Houseblock 7"
+      )
+    ).thenReturn(result)
+
+    val response = mockMvc.get("/prisons/MDI/location-prefix") {
+      param("groupName", "Houseblock 7")
+    }
+      .andDo { print() }
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationPrefixControllerTest.kt:37]: print?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationPrefixControllerTest.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    verify(locationService).getLocationPrefixFromGroup(
+      "MDI", "Houseblock 7",
+    )
+  }
+
+  @Test
+  fun `400 response when group name missing`() {
+    mockMvc.get("/prisons/MDI/location-prefix")
+      .andDo { print() }
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationPrefixControllerTest.kt:101]: print?

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/LocationPrefixIntegrationTest.kt (hunk 2 above)
Comment (markford-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/LocationPrefixIntegrationTest.kt:24]: Just lifted from whereabouts

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationPrefixControllerTest.kt (hunk 3 above)
Comment (markford-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationPrefixControllerTest.kt:37]: its just a test - will take it out tho

**Pull request merged**